app/controllers/admin/admin_users_controller.rb

```diff
@@ -39,76 +39,115 @@ def update_user
         flash[:error] = ts('There was an error updating user %{name}', :name => params[:id])
         redirect_to(request.env["HTTP_REFERER"] || root_path)
       end
-    elsif params[:admin_action]
-      @user = User.find_by_login(params[:user_login])
+    else
       @admin_note = params[:admin_note]
-      if @admin_note.blank?
-        flash[:error] = ts("You must include notes in order to perform this action")
-        redirect_to(request.env["HTTP_REFERER"] || root_path)
+      @user = User.find_by_login(params[:user_login])
+      submitted_kin_user = User.find_by_login(params[:next_of_kin_name])
+ 
+      # there is a next of kin username, but no email
+      if params[:next_of_kin_name].present? && params[:next_of_kin_email].blank?
+        flash[:error] = ts('Fannish next of kin email is missing.')
+        redirect_to request.referer || root_path
+
+      # there is a next of kin email, but no username
+      elsif params[:next_of_kin_name].blank? && params[:next_of_kin_email].present?
+        flash[:error] = ts('Fannish next of kin user is missing.')
+        redirect_to request.referer || root_path
+
+      # there is a next of kin username, but it is not a valid user
+      elsif params[:next_of_kin_name].present? && submitted_kin_user.nil?
+        flash[:error] = ts('Fannish next of kin user is invalid.')
+        redirect_to request.referer || root_path
+
+      # there is an admin action selected, but no note entered  
+      elsif params[:admin_action].present? && @admin_note.blank?
+        flash[:error] = ts('You must include notes in order to perform this action.')
```

Comment:
Line is too long. [85/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.